doc/api/brotli.md

```diff
@@ -0,0 +1,404 @@
+**Default:** `false`
+* `sizeHint` {integer} (compression only)
+**Default:** `0`
+* `npostfix` {integer} (compression only)
+**Default:** `0`
+* `ndirect` {integer} (compression only)
+**Default:** `0`
+* `disableRingBufferReallocation` {boolean} (decompression only)
+**Default:** `false`
+* `largeWindow` {boolean}
+**Default:** `false`
+* `info` {boolean} (If `true`, returns an object with `buffer` and `engine`)
+
+See [`brotli/encode.h`][] and [`brotli/decode.h`][] for more information on
+these.
+
+## Class: brotli.Brotli
+
+Not exported by the `brotli` module. It is documented here because it is the
+base class of the compressor/decompressor classes.
+
+### brotli.bytesWritten
+
+* {number}
+
+The `brotli.bytesWritten` property specifies the number of bytes written to
+the engine, before the bytes are processed (compressed or decompressed,
+as appropriate for the derived class).
+
+### brotli.close([callback])
+
+Close the underlying handle.
+
+### brotli.flush(callback)
+
+Flush pending data. Don't call this frivolously, premature flushes negatively
+impact the effectiveness of the compression algorithm.
+
+Calling this only flushes data from the internal `brotli` state, and does not
+perform flushing of any kind on the streams level. Rather, it behaves like a
+normal call to `.write()`, i.e. it will be queued up behind other pending
+writes and will only produce output when data is being read from the stream.
+
+### brotli.setParameter(param, value)
+
+- `param` {number} A valid parameter from [Constants][]
+- `value` {number}
+
+Dynamically set a parameter.
+
+### brotli.reset()
+
+Reset the compressor/decompressor to factory defaults. Only applicable to
+the inflate and deflate algorithms.
+
+## Class: brotli.Compress
+
+Compress a stream.
+
+### new brotli.Compress(options)
+
+Creates and returns a new [Compress][] object with the given [`options`][].
+
+### Class Property: brotli.Compress.codes
+
+Error codes for compression operations.
+
+## Class: brotli.Decompress
+
+Decompress a stream.
+
+### new brotli.Decompress(options)
+
+Creates and returns a new [Decompress][] object with the given [`options`][].
+
+### Class Property: brotli.Decompress.codes
+
+Error codes for decompression operations.
+
+## brotli.constants
+
+Provides an object enumerating Brotli-related constants.
+
+## Convenience Methods
+
+<!--type=misc-->
+
+All of these take a [`Buffer`][], [`TypedArray`][], [`DataView`][],
+[`ArrayBuffer`][] or string as the first argument, an optional second argument
+to supply options to the `brotli` classes and will call the supplied callback
+with `callback(error, result)`.
+
+Every method has a `*Sync` counterpart, which accept the same arguments, but
+without a callback.
+
+### brotli.compress(buffer[, options][, callback])
+
+- `buffer` {Buffer|TypedArray|DataView|ArrayBuffer|string}
+
+Decompress a chunk of data with [Decompress][].
+If callback is omitted a Promise will be returned.
```

Comment:
`` Promise `` -> `` `Promise` `` or even `{Promise}` (will be linkified)?
Ditto below.